modules/ppcp-button/resources/js/modules/Renderer/CreditCardRenderer.js

```diff
@@ -25,141 +24,270 @@ class CreditCardRenderer {
+                hostedFields.on('cardTypeChange', (event) => {
+                    if (!event.cards.length) {
+                        this.cardValid = false;
+                        return;
+                    }
+                    const validCards = this.defaultConfig.hosted_fields.valid_cards;
+                    this.cardValid = validCards.indexOf(event.cards[0].type) !== -1;
+
+                    const className = this._cardNumberFiledCLassNameByCardType(event.cards[0].type);
+                    this._recreateElementClassAttribute(cardNumber, cardNumberField.className);
+                    if (event.cards.length === 1) {
+                        cardNumber.classList.add(className);
+                    }
+                })
+                hostedFields.on('validityChange', (event) => {
+                    this.formValid = Object.keys(event.fields).every(function (key) {
+                        return event.fields[key].isValid;
+                    });
+                });
+                hostedFields.on('empty', (event) => {
+                    this._recreateElementClassAttribute(cardNumber, cardNumberField.className);
+                    this.emptyFields.add(event.emittedBy);
+                });
+                hostedFields.on('notEmpty', (event) => {
+                    this.emptyFields.delete(event.emittedBy);
+                });
+
+                show(buttonSelector);
+
+                if (document.querySelector(wrapper).getAttribute('data-ppcp-subscribed') !== true) {
+                    document.querySelector(buttonSelector).addEventListener(
+                        'click',
+                        event => {
+                            event.preventDefault();
+                            this._submit(contextConfig);
+                        }
+                    );
+
+                    document.querySelector(wrapper).setAttribute('data-ppcp-subscribed', true);
+                }
             });
-            hostedFields.on('cardTypeChange', (event) => {
-                if ( ! event.cards.length ) {
-                    this.cardValid = false;
-                    return;
+
+            document.querySelector('#payment_method_ppcp-credit-card-gateway').addEventListener(
+                'click',
+                () => {
+                    document.querySelector('label[for=ppcp-credit-card-gateway-card-number]').click();
                 }
-                const validCards = this.defaultConfig.hosted_fields.valid_cards;
-                this.cardValid = validCards.indexOf(event.cards[0].type) !== -1;
+            )
 
-                const className = this._cardNumberFiledCLassNameByCardType(event.cards[0].type);
-                this._recreateElementClassAttribute(cardNumber, cardNumberField.className);
-                if (event.cards.length === 1) {
-                    cardNumber.classList.add(className);
+            return;
+        }
+
+        if (typeof paypal.CardFields !== 'undefined') {
```

Comment:
Should it be in the same file with the old ACDC? Seems like nothing is reused.

An easy way to separate could be just to create a new class with the same interface (`render` function) and decide which one to instantiate in `button.js`